### PR TITLE
[Form] Handle false as empty value on expanded choices

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/issues/31572
| License       | MIT
| Doc PR        | -

This is the 3.4 version of https://github.com/symfony/symfony/pull/32747. The tests are the same. The added code has to be removed from master (if accepted).

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/CheckboxType.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         // doing so also calls setDataLocked(true).
         $builder->setData(isset($options['data']) ? $options['data'] : false);
         $builder->addViewTransformer(new BooleanToStringTransformer($options['value']));
+        $builder->setAttribute('false_is_empty', true);
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Core/Type/CheckboxType.php]: `_false_is_empty`. I would also add a comment to explain why we add it there, and that it not needed in 5.1 either.

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php [hunk 2]
```diff
@@ -2046,4 +2046,45 @@ public function provideTrimCases()
             'Multiple expanded' => [true, true],
         ];
     }
+
+    /**
+     * @dataProvider expandedIsEmptyWhenNoRealChoiceIsSelectedProvider
+     */
+    public function testExpandedIsEmptyWhenNoRealChoiceIsSelected($expected, $submittedData, $multiple, $required, $placeholder)
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: What about a test in the the `CheckboxTypeTest` too?

**Review by HeahDude — COMMENTED** [review]
Looks good, thanks for fixing this. Some minor comments

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php (hunk 2 above)
Comment (fancyweb) [reply src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: I added a small test in `CheckboxTypeTest` to be defensive.

**Review by xabbuh — APPROVED** [review]

**Review by HeahDude — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**